### PR TITLE
Adhere to Solidity style guide and update comments

### PR DESCRIPTION
**Pull request by ilikesymmetry**
Change reasonings annotated in-line.

### CODE REVIEW

File: src/SpendPermissionManager.sol [hunk 1]
```diff
@@ -1,166 +1,197 @@
...
-import {IERC721} from "openzeppelin-contracts/contracts/token/ERC721/IERC721.sol";
-import {MagicSpend} from "magic-spend/MagicSpend.sol";
 import {SafeERC20} from "openzeppelin-contracts/contracts/token/ERC20/utils/SafeERC20.sol";
-import {SafeTransferLib} from "solady/utils/SafeTransferLib.sol";
-import {IERC165} from "openzeppelin-contracts/contracts/interfaces/IERC165.sol";
+import {IERC721} from "openzeppelin-contracts/contracts/token/ERC721/IERC721.sol";
 import {ERC165Checker} from "openzeppelin-contracts/contracts/utils/introspection/ERC165Checker.sol";
+import {CoinbaseSmartWallet} from "smart-wallet/CoinbaseSmartWallet.sol";
+import {EIP712} from "solady/utils/EIP712.sol";
+import {SafeTransferLib} from "solady/utils/SafeTransferLib.sol";
```
Comment (ilikesymmetry) [inline src/SpendPermissionManager.sol:13]: updated import sorting

File: src/SpendPermissionManager.sol [hunk 2]
```diff
@@ -1,166 +1,197 @@
...
-    /// @dev Start time of the period (unix seconds).
-    uint48 start;
-    /// @dev End time of the period (unix seconds).
-    uint48 end;
-    /// @dev Accumulated spend amount for period.
-    uint160 spend;
-}
-
 /// @title SpendPermissionManager
+///
```
Comment (ilikesymmetry) [inline src/SpendPermissionManager.sol:18]: Coinbase style guide requires `///` newlines for multiple natspec tag types

File: src/SpendPermissionManager.sol [hunk 3]
```diff
@@ -1,166 +1,197 @@
...
+
+    /// @notice Separated contract for validating signatures and executing ERC-6492 side effects.
     PublicERC6492Validator public immutable PUBLIC_ERC6492_VALIDATOR;
 
     /// @notice MagicSpend singleton (https://github.com/coinbase/magic-spend).
     address public immutable MAGIC_SPEND;
 
-    bytes32 public constant PERMISSION_TYPEHASH = keccak256(
+    /// @notice EIP-712 hash of SpendPermission type.
+    bytes32 public constant SPEND_PERMISSION_TYPEHASH = keccak256(
```
Comment (ilikesymmetry) [inline src/SpendPermissionManager.sol:98]: missing natspecs for typehashes

File: src/SpendPermissionManager.sol [hunk 4]
```diff
@@ -1,166 +1,197 @@
...
-    uint256 transient private _expectedReceiveAmount;
+    /// @notice Spend permission is approved.
+    mapping(bytes32 hash => bool approved) internal _isApproved;
 
     /// @notice Spend permission is revoked.
     mapping(bytes32 hash => bool revoked) internal _isRevoked;
 
-    /// @notice Spend permission is approved.
-    mapping(bytes32 hash => bool approved) internal _isApproved;
-
```
Comment (ilikesymmetry) [inline src/SpendPermissionManager.sol:120]: for consistency with non-view functions, moving approve ahead of revoke

File: src/SpendPermissionManager.sol [hunk 5]
```diff
@@ -1,166 +1,197 @@
...
     /// @notice Last updated period for a spend permission.
     mapping(bytes32 hash => PeriodSpend) internal _lastUpdatedPeriod;
 
+    /// @notice A flag to indicate if the contract can receive native token transfers, and the expected amount.
+    ///
+    /// @dev Contract can only receive exactly the expected amount during the execution of `spend` for native tokens.
+    uint256 private _expectedReceiveAmount;
```
Comment (ilikesymmetry) [inline src/SpendPermissionManager.sol:129]: private variables should follow internal by visibility ordering

File: src/SpendPermissionManager.sol [hunk 6]
```diff
@@ -1,166 +1,197 @@
...
+
+    /// @notice Spend permission was approved.
+    ///
+    /// @param hash Unique hash representing the spend permission.
+    /// @param spendPermission Details of the spend permission.
+    event SpendPermissionApproved(bytes32 indexed hash, SpendPermission spendPermission);
```
Comment (ilikesymmetry) [inline src/SpendPermissionManager.sol:135]: events should follow storage and precede errors

File: src/SpendPermissionManager.sol [hunk 7]
```diff
@@ -204,70 +235,44 @@ contract SpendPermissionManager is EIP712 {
...
-    /// @param account Account that had its tokens spent via a spend permission.
-    /// @param spender Entity that spent `account`'s tokens.
-    /// @param token Address of token spent via a spend permission.
-    /// @param periodSpend Start and end of the current period with marginal new spend (struct).
-    event SpendPermissionUsed(
-        bytes32 indexed hash, address indexed account, address indexed spender, address token, PeriodSpend periodSpend
-    );
+    /// @notice Require a specific sender for an external call.
+    /// @param sender Expected sender for call to be valid.
+    modifier requireSender(address sender) {
```
Comment (ilikesymmetry) [inline src/SpendPermissionManager.sol:249]: modifiers should precede all functions

File: src/SpendPermissionManager.sol [hunk 8]
```diff
@@ -352,45 +356,45 @@ contract SpendPermissionManager is EIP712 {
...
-    /// @dev The `account` of the permissions must match, but the remaining fields can differ.
-    /// @dev Can only be called by the `account` of a permission.
+    /// @dev Comparing the revoked spend permission's last updated period mitigates frontrunning with further spend.
+    /// @dev The accounts of the spend permissions must match, but all other fields can differ.
+    /// @dev Reverts if not called by the account of the spend permissions.
     ///
     /// @param permissionToApprove Details of the spend permission to approve.
     /// @param permissionToRevoke Details of the spend permission to revoke.
-    /// @param lastValidUpdatedPeriod Last valid updated period for the spend permission being revoked.
+    /// @param expectedLastUpdatedPeriod Expected last updated period for the spend permission being revoked.
```
Comment (ilikesymmetry) [inline src/SpendPermissionManager.sol:367]: not official, but following our existing convention for "expected" comparison in errors

File: src/SpendPermissionManager.sol [hunk 9]
```diff
@@ -352,45 +356,45 @@ contract SpendPermissionManager is EIP712 {
...
+
         // revoke old and approve new spend permissions
         _revoke(permissionToRevoke);
         return _approve(permissionToApprove);
     }
 
     /// @notice Revoke a spend permission to disable its use indefinitely.
     ///
-    /// @dev Can only be called by the `account` of a permission.
+    /// @dev Reverts if not called by the account of the spend permission.
```
Comment (ilikesymmetry) [inline src/SpendPermissionManager.sol:397]: not official, but we use "reverts if" convention elsewhere so unifying

File: src/SpendPermissionManager.sol [hunk 10]
```diff
@@ -450,10 +455,8 @@ contract SpendPermissionManager is EIP712 {
             revert SpendValueWithdrawAmountMismatch(value, withdrawRequest.amount);
         }
 
-        // Get the hash of the spend permission
+        // check WithdrawRequest nonce suffix matches spend permission hash suffix.
```
Comment (ilikesymmetry) [inline src/SpendPermissionManager.sol:458]: double check my language here please @amiecorso!

File: src/SpendPermissionManager.sol [hunk 11]
```diff
@@ -468,29 +471,29 @@ contract SpendPermissionManager is EIP712 {
         _transferFrom(spendPermission.token, spendPermission.account, spendPermission.spender, value);
     }
 
-    /// @notice Check if a spend permission is revoked.
+    /// @notice Get if a spend permission is approved.
```
Comment (ilikesymmetry) [inline src/SpendPermissionManager.sol:474]: not officially documented, but our use of "check" is paired with a conditional and potential revert, so unified with our existing convention of "get" for other view functions.

File: src/SpendPermissionManager.sol [hunk 12]
```diff
@@ -468,29 +471,29 @@ contract SpendPermissionManager is EIP712 {
...
     ///
     /// @param spendPermission Details of the spend permission.
     ///
-    /// @return revoked True if spend permission is revoked.
-    function isRevoked(SpendPermission memory spendPermission) public view returns (bool) {
-        return _isRevoked[getHash(spendPermission)];
+    /// @return approved True if spend permission is approved.
+    function isApproved(SpendPermission memory spendPermission) public view returns (bool) {
+        return _isApproved[getHash(spendPermission)];
```
Comment (ilikesymmetry) [inline src/SpendPermissionManager.sol:480]: diff looks messed up here, but just swapped order of `isApproved` ahead of `isRevoked` for parity with non-view functions

File: src/SpendPermissionManager.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (amiecorso) [reply src/SpendPermissionManager.sol:13]: oh interesting -- the coinbase solidity style guide indicates that imports should be alphabetical, but after grouping by internal v.s. external.  solidity style guide doesn't seem to specify anything other than they should be placed at the top of the file.

File: src/SpendPermissionManager.sol [hunk 13]
```diff
@@ -1,166 +1,197 @@
...
 import {ERC165Checker} from "openzeppelin-contracts/contracts/utils/introspection/ERC165Checker.sol";
+import {CoinbaseSmartWallet} from "smart-wallet/CoinbaseSmartWallet.sol";
+import {EIP712} from "solady/utils/EIP712.sol";
+import {SafeTransferLib} from "solady/utils/SafeTransferLib.sol";
 
 import {PublicERC6492Validator} from "./PublicERC6492Validator.sol";
 
-
-/// @notice A spend permission for an external entity to be able to spend an account's tokens.
-struct SpendPermission {
```
[the remark this answers is not in the document]
Comment (amiecorso) [reply src/SpendPermissionManager.sol:19]: ah yeah... I see that in the coinbase style guide now

[the remark this answers is not in the document]
Comment (amiecorso) [reply src/SpendPermissionManager.sol:19]: that's ok 😂 I really appreciate this style pass!

File: src/SpendPermissionManager.sol (hunk 10 above)
Comment (amiecorso) [reply src/SpendPermissionManager.sol:458]: this is great, much clearer than "hash portion matches expected" haha

**Review by amiecorso — APPROVED** [review]
great cleanup! I left a question about the imports but otherwise I think the language changes and style nits are improvements and I'm aligned. thanks for this!

**Pull request merged**